backend/tournesol/views/polls.py

```diff
@@ -295,6 +295,13 @@ def annotate_and_prefetch_scores(self, queryset, request, poll: Poll):
         criteria_weight = self._build_criteria_weight_condition(
             request, poll, when="all_criteria_scores__criteria"
         )
+
+        # FIXME: we can significantly improve the performance of the queryset
+        # by filtering the criteria on their names, to remove those that are
+        # not present in the request.
+        #
+        #   ex:
+        #       all_criteria_scores__criteria__in=[...]
```

Comment:
Can we move this suggestion in a ticket instead of a comment?

An other good optimization would be that the route used 95% of the time simply reads the tournesol_score on EntityPollRating (i.e. the main criteria) without doing the join with the criteria scores.

Only when users specify other criteria or want to weight each criteria manually, we'd use this route.

Reply:
See: https://github.com/tournesol-app/tournesol/issues/1881